src/transformers/models/falcon_mamba/modeling_falcon_mamba.py

```diff
@@ -733,6 +757,10 @@ def prepare_inputs_for_generation(
                 )
             if cache_position[0] > 0:
                 input_ids = input_ids[:, -1].unsqueeze(-1)
+
+                if attention_mask is not None:
+                    attention_mask = attention_mask[:, -1].unsqueeze(-1)
```

Comment:
```suggestion
                    attention_mask = None
```
actually we should not even pass a mask as it is always gonna be `1` here no?

Reply:
Good point !